srcs/incs/Utils.hpp

```diff
@@ -0,0 +1,32 @@
+#ifndef SRCS_INCS_UTILS_HPP_
+#define SRCS_INCS_UTILS_HPP_
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <unistd.h>
+#include <algorithm>
+#include <cstring>
+#include <ctime>
+#include <sstream>
+#include <string>
+#include <MimeTypes.hpp>
+
+std::string	TrimString(const std::string &str, const std::string &trim_chars);
+std::string	ToLowerString(std::string str);
+char		*DuplicateString(const std::string &str);
+std::string	DecodeUrl(const std::string &encoded_url);
+bool		IsExecutable(const std::string &path);
+bool		IsValidPath(const std::string &path);
+bool		IsDirectory(const std::string &path);
+bool		IsRegularFile(const std::string &path);
+std::string	PathExtension(const std::string &path);
+std::string	GetMimeType(const std::string &file_path);
```

Comment:
We have a bit of duplication between this and File.hpp, don't we?

Reply:
Doing these:
```
diff --git a/srcs/incs/IRequest.hpp b/srcs/incs/IRequest.hpp
index db19f80..4713646 100644
--- a/srcs/incs/IRequest.hpp
+++ b/srcs/incs/IRequest.hpp
@@ -12,7 +12,6 @@ class IRequest {
 		virtual void		Reset() = 0;
 		virtual bool		IsPartial() const = 0;
 		virtual bool		IsComplete() const = 0;
-		virtual bool		IsInvalid() const = 0;
 };
 
 #endif  // SRCS_INCS_IREQUEST_HPP_
diff --git a/srcs/incs/SyscallWrap.hpp b/srcs/incs/SyscallWrap.hpp
index 7ae768b..9b2fbea 100644
--- a/srcs/incs/SyscallWrap.hpp
+++ b/srcs/incs/SyscallWrap.hpp
@@ -4,8 +4,6 @@
 
 class SyscallWrap {
 	public:
-		static ssize_t writeWr(int fd, const void *buf, size_t count);
-		static ssize_t readWr(int fd, void *buf, size_t count);
 		static int pipeWr(int pipefd[2]);
 		static int dupWr(int oldfd);
 		static int dup2Wr(int oldfd, int newfd);
@@ -13,7 +11,6 @@ class SyscallWrap {
 						   char *const envp[]);
 		static int closeWr(int fd);
 		static pid_t forkWr(void);
-		static pid_t waitpidWr(pid_t pid, int *wstatus, int options);
 };
 
 #endif  // SRCS_INCS_SYSCALLWRAP_HPP_
diff --git a/srcs/incs/Utils.hpp b/srcs/incs/Utils.hpp
index 0f2aefc..ef2acae 100644
--- a/srcs/incs/Utils.hpp
+++ b/srcs/incs/Utils.hpp
@@ -14,13 +14,8 @@ std::string	TrimString(const std::string &str, const std::string &trim_chars);
 std::string	ToLowerString(std::string str);
 char		*DuplicateString(const std::string &str);
 std::string	DecodeUrl(const std::string &encoded_url);
-bool		IsExecutable(const std::string &path);
-bool		IsValidPath(const std::string &path);
-bool		IsDirectory(const std::string &path);
-bool		IsRegularFile(const std::string &path);
+bool		IsExecutable(const std::string &path);;
 std::string	PathExtension(const std::string &path);
-std::string	GetMimeType(const std::string &file_path);
-std::string	CurrentDate();
 
 template <typename T>
 std::string	ValueToString(const T &value) {
```
doesn't break anything, AFAIK

Reply:
Yes, there is a bit of duplication. My suggestion would be to call these functions from the `File` class.

For example:
```
void	File::SetPathExtension_() {
	path_extension_ = PathExtension(file_path_);
}
```

This way, we can access them anywhere or reuse them in other places where it might not make sense to create a `File` class.

Just an idea, what do we think about this?

Reply:
I like it. We still had to change the name of the File class in the first place anyway, didn't we?

Reply:
I've removed `CurrentDate()` and `GetMimeTypes()`

While testing the PHP CGI with WordPress, I found a bug. It happens when the index is a CGI file, for example, index.php.

And we could use some of these path functions in the future to fix it.

Reply:
I've created an issue for this, https://github.com/42AGV/webserv/issues/117